### PR TITLE
V156 admin index dashboard strict

### PR DESCRIPTION
**Pull request by mc12345678**
Address notifications made when using admin/index_dashboard.php while in strict mode.

### CODE REVIEW

File: admin/index_dashboard.php [hunk 1]
```diff
@@ -342,24 +345,24 @@ function drawCharts() {
...
-          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_yellow.gif'); ?>&nbsp;&nbsp;<?php echo $user_array[1]; ?></div>
-          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_red.gif'); ?>&nbsp;&nbsp;<?php echo $user_array[2]; ?></div>
-          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_red_light.gif'); ?>&nbsp;&nbsp;<?php echo $user_array[3]; ?></div>&nbsp;
+          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_green.gif'); ?>&nbsp;&nbsp;<?php echo (!empty($user_array[0]) ? $user_array[0] : ''); ?></div>
+          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_yellow.gif'); ?>&nbsp;&nbsp;<?php echo (!empty($user_array[1]) ? $user_array[1] : ''); ?></div>
+          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_red.gif'); ?>&nbsp;&nbsp;<?php echo (!empty($user_array[2]) ? $user_array[2] : ''); ?></div>
+          <div class="col"><?php echo zen_image(DIR_WS_IMAGES . 'icon_status_red_light.gif'); ?>&nbsp;&nbsp;<?php echo (!empty($user_array[3]) ? $user_array[3] : ''); ?></div>&nbsp;
```
Comment (drbyte) [inline admin/index_dashboard.php]: Since these arrays are initialized earlier, could all these be shortened using the `?:` syntax?

`echo $user_array[0] ?: '';`

**Comment by mc12345678** [comment]
Unfortunately, while there is a possibility that each of the four sub-keys have been initialized, there is no guarantee that all or any more than one will be initialized, but the four states of each of the variables will be attempted. Perhaps on a busy store all of them will be defined, but not so on one that is somewhat idle.

Further, unfortunately using the "Elvis operator" as suggested would still result in a strict notification in the case(s) where the status was not detected. And the null coalesce operator while it would resolve the issue isn't available until php 7.x. Unless there was some sort of ditch out process to not cause an error for someone using what is reported to be an allowable php version (especially on the first page of the admin) then to "simplify" the code something else needs to be done. Either all possible status settings need to purposefully be stepped through to ensure that they are all set to 0 before attempting to add to them and then later display can be more "straight forward" or some level of filtering amenable to the supported software should be incorporated/kept. 

Basically, if all of the arrays are cycled through for each possible status and given a value then yes the "Elvis operator" as suggested would work for php 5.3 and above.

**Comment by drbyte** [comment]
Agreed. Given that there are just 4 status keys (0-3) to initialize for each, let's just initialize them all at the start.
IMO initializing early is preferable, since:
the initialization step is all in one spot, semantic, and obvious to the reader/maintainer of the code, 
and then later output logic is also obvious, sensible and as brief as possible while being as explicit as necessary.

(I agree, null-coalesce would be more desirable, but we can't ... yet)

For extreme brevity for the init, what do you think of this? ...
```php
// initialize
$user_array = $guest_array = $spider_array = [0=>0, 1=>0, 2=>0, 3=>0];
```

**Comment by mc12345678** [comment]
Init looks good: forces a 0 based key, all known keys set to 0 and all arrays initialized. Works.
